cdb2jdbc/src/main/java/com/bloomberg/comdb2/jdbc/Comdb2Handle.java

```diff
@@ -99,6 +99,7 @@
     private int snapshotOffset;
     private boolean isHASql = false;
     private int isRetry;
+    private int nSetsQuerySent;
```

Comment:
Keep track of the number set commands that we have sent to the replicant that we are connected to.

Reply:
Kyle was managing to coerce the JDBC driver into *failing* to send set directives.  The result was that alot of his failing tests were failing because the database never received 'set transaction serializable'.  It seemed like this was occurring because the code would issue these set commands before reading the results of a previous operation.  When the code eventually *did* read the results, it got a LASTROW, and set nSetsSent to sets.size(), even though the just issued set commands had not ever been sent.

The fix is to introduce a new variable, nSetsQuerySent.  This is set to be the current sets.size() when we issue a sendQuery (and actually SEND the set commands).  When we receive a LASTROW, the code updates nSetsSent to the most recent value of nSetsQuerySent (what we've actually sent to the database) rather than the current sets.size().